### PR TITLE
Add transfer-files command support

### PR DESCRIPTION
**Pull request by RobiNino**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
I think we should simulate the following scenario - 
During phase 1 or phase 2, jfrog-cli looses connectivity with the source Artifactory instance. The source instance will simply not recieves the chunks. JFrog CLI in this case will just log the error and move on to the next chunk. The chunks that failed to be sent will never be sent again.

File: artifactory/commands/transferfiles/api.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+type NodeIdResponse struct {
+	NodeId string `json:"node_id,omitempty"`
+}
+
+type UuidTokenResponse struct {
+	UuidToken string `json:"uuid_token,omitempty"`
+}
+
+// Fill tokens batch till full. Return if no new tokens are available.
```
[the remark this answers is not in the document]
Comment (RobiNino) [reply artifactory/commands/transferfiles/api.go:101]: Added description above polIUploads

File: artifactory/commands/transferfiles/migration.go [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+func (m *migrationPhase) setSourceDetails(details *coreConfig.ServerDetails) {
+	m.srcRtDetails = details
+}
+
+func (m *migrationPhase) setTargetDetails(details *coreConfig.ServerDetails) {
+	m.targetRtDetails = details
+}
+
+func (m *migrationPhase) run() error {
```
[the remark this answers is not in the document]
Comment (RobiNino) [reply artifactory/commands/transferfiles/migration.go:123]: As discussed, will look into it in future PRs

File: utils/log/log.go [hunk 3]
```diff
@@ -34,3 +39,31 @@ func getJfrogCliLogTimestamp() int {
 func SetDefaultLogger() {
 	log.SetLogger(log.NewLoggerWithFlags(GetCliLogLevel(), nil, getJfrogCliLogTimestamp()))
 }
+
+func CreateLogFile() (*os.File, error) {
```
[the remark this answers is not in the document]
Comment (gailazar300) [reply utils/log/log.go:43]: This is the existing logging mechanism - I moved those functions from the cli to core cause the 'transfer progress bar" was moved to core and used this functionality.

File: artifactory/commands/transferfiles/transfer.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+	srcRepos, err := tdc.getSrcLocalRepositories()
+	if err != nil {
+		return err
+	}
+
+	targetRepos, err := tdc.getTargetLocalRepositories()
+	if err != nil {
+		return err
+	}
```
[the remark this answers is not in the document]
Comment (RobiNino) [reply artifactory/commands/transferfiles/transfer.go:84]: We check it for every repo separately in `verifyRepoExistsInTarget`

File: artifactory/commands/transferfiles/utils.go [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+	if targetAuth.TargetToken == "" {
+		targetAuth.TargetUsername = targetRtDetails.User
+		targetAuth.TargetPassword = targetRtDetails.Password
+	}
+	return targetAuth
+}
+
+// This variable holds the number of upload chunk that was sent to the user plugin to process.
+// Together with this mutex, they control the load on the user plugin and couple it to the local number of threads.
+var curProcessedUploadChunks = 0
```
[the remark this answers is not in the document]
Comment (RobiNino) [reply artifactory/commands/transferfiles/utils.go:90]: Will do as part of the structure change of the phases in future PRs

File: artifactory/commands/transferfiles/utils.go [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+	for i := 0; i < len(uuidTokens); i++ {
+		if token == uuidTokens[i] {
+			return append(uuidTokens[:i], uuidTokens[i+1:]...)
+		}
+	}
+	log.Error("Unexpected uuid token found: " + token)
+	return uuidTokens
+}
+
+func handleFilesOfCompletedChunk(chunkFiles []FileUploadStatusResponse) {
```
[the remark this answers is not in the document]
Comment (RobiNino) [reply artifactory/commands/transferfiles/utils.go:153]: @gailazar300 's commit which was not included in this PR involves in this area.

File: artifactory/commands/transferfiles/utils.go [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+		return []string{}, err
+	}
+
+	filesNames, err := biUtils.ListFiles(dirPath, false)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, file := range filesNames {
+		if strings.HasPrefix(filepath.Base(file), repoKey) {
```
[the remark this answers is not in the document]
Comment (RobiNino) [reply artifactory/commands/transferfiles/utils.go:274]: Noted. Will replace with regexp later on

**Pull request merged**